src/main/java/heronarts/lx/midi/surface/APC40Mk2.java

```diff
@@ -265,6 +290,7 @@ void registerChannel(LXBus channel) {
       } else {
         register(null);
       }
+      updateLCPKnobs();
```

Comment:
It feels like this should be handled within register() as it loops over knob construction.. Is it having another side effect that I'm not noticing here?

Reply:
It needs to be callable elsewhere, too: When shift, bankLeft, or bankRight are pressed, we need to re-light the knobs according to the underlying value they newly control.